### PR TITLE
Classes - movie description extension

### PR DESCRIPTION
**Pull request by Rahulkug**
## Checklist
- Solution is tested on sphere engine <br/>
    `Note:` Please copy only the solution part. Do not copy/paste the template code. That way, any issues with template code can be caught early on.
    - [ ] Author
    - [x] Reviewer
- PR title is correct - [PR title conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#title-for-a-pull-request)
    - [ ] Author
    - [x] Reviewer
- Folder/File naming is correct - [Naming conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#naming-of-files-and-folders)
    - [ ] Author
    - [x] Reviewer
- Coding conventions are followed - [Coding conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#coding-conventions)
    - [ ] Author
    - [x] Reviewer
- Test cases (action item for **reviewer**) <br/>
    - To be ensured by checking the problem on sphere-engine
    - If any test cases are missing, please add them - [how to add new test cases](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#adding-new-test-cases)
    - [x] There are at least 2 test cases
    - [x] There is at least 1 test case that is different from the sample test cases

### CODE REVIEW

**Review by satya-9 — APPROVED** [review]

File: solutions/classes/movie-description-extension.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+function readLine() {
+  idx++;
+  return data[idx - 1];
+}
+
+// -------- Do NOT edit anything above this line ----------
+class Movie{
+  constructor(
+    lengthInminutes = 120,
```
Comment (sailesh-ramayanam) [inline solutions/classes/movie-description-extension.js:14]: The problem description does not talk about default values. Please remove them.

File: solutions/classes/movie-description-extension.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    text = text + this.numCharacters + " characters, is ";
+    text = text + this.lengthInminutes + " minutes long, and is ";
+    text = text + flag + "certified.";
+    return text;
+  }
+};
+
+let language = readLine();
+let numCharacters = readLine();
+let lengthInminutes = readLine();
```
Comment (sailesh-ramayanam) [inline solutions/classes/movie-description-extension.js:45]: `numCharacters` and `lengthInminutes` need to be converted to integers.

**Review by sailesh-ramayanam — CHANGES_REQUESTED** [review]
@satya-9 Please make the code changes as suggested in the comments and description change as suggested below.
**Modify the description as below.**
Recall the `Movie` class you defined in one of the previous problems. Its description is given below for your reference.
### Previous description
Write a Movie class for which you can create movie objects. The objects should have the following attributes: `lengthInMinutes`, `numCharacters`, `language`. The class should have a run method which prints out: "This is a `language` movie with `numCharacters` characters and is `lengthInMinutes` minutes long."

Now, we want to enhance the `Movie` class by adding a method to certify the movie.
### Current description

1. Enhance the `Movie` class by adding a method `certify`, which returns `true` or `false`. It should return `true` if the movie has at least 2 characters and is less than or equal to 4 hours long (which means the movie is certified). Otherwise, it should return `false`.
2. Modify the run message to include the certification.
    - If certified then the message should be "This is a `language` movie with `numCharacters` characters, is `lengthInMinutes` minutes long and is certified."
    - If not certified then the message should be "This is a `language` movie with `numCharacters` characters, is `lengthInMinutes` minutes long and is not certified."
 
**Modify the output format as below**
- If certified then the output should be "This is a `language` movie with `numCharacters` characters, is `lengthInMinutes` minutes long and is certified."
- If not certified then the output should be "This is a `language` movie with `numCharacters` characters, is `lengthInMinutes` minutes long and is not certified."
You need not worry about input/output. It is handled by the template provided. You only need to implement the `Movie` class.

**Test cases** 
Format of a test case should be Input, Output, Explanation.
1. All the sample inputs are given together and all the sample outputs are given together. Please segregate then as Test case 1 and Test case 2.
2. No need to give previous test case.
3. Provide explanation to all the test cases.

File: solutions/classes/movie-description-extension.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+data = data.split("\n");
+
+function readLine() {
+  idx++;
+  return data[idx - 1];
+}
+
+// -------- Do NOT edit anything above this line ----------
+class Movie {
+  constructor() {
```
Comment (sailesh-ramayanam) [inline solutions/classes/movie-description-extension.js:13]: @satya-9 Input parameters are missing.

**Comment by sailesh-ramayanam** [comment]
@satya-9 
1. Formatting is missing in Current Description - for terms like Movie, certify, true, false (please refer to my previous comment for formatting).
2. Explanation
    - Test case 1: The movie name has 0 number of characters which is less than 2. So, it is not certified.
    - Test case 2: The movie is 400 minutes long (which is 6 hrs 40 minutes) greater than 4 hrs. So, it is not certified.

File: solutions/classes/movie-description-extension.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    text = text + this.language + " movie with ";
+    text = text + this.numCharacters + " characters, is ";
+    text = text + this.lengthInMinutes + " minutes long, and is ";
+    text = text + flag + "certified.";
+    return text;
+  }
+};
+
+let language = readLine();
+let numCharacters = parseInt(readLine());
```
Comment (sailesh-ramayanam) [inline solutions/classes/movie-description-extension.js:40]: @satya-9 `parseInt` is missing in the widget (for the line below as well).

**Comment by satya-9** [comment]
> @satya-9
> 
> 1. Formatting is missing in Current Description - for terms like Movie, certify, true, false (please refer to my previous comment for formatting).
> 2. Explanation
>    
>    * Test case 1: The movie name has 0 number of characters which is less than 2. So, it is not certified.
>    * Test case 2: The movie is 400 minutes long (which is 6 hrs 40 minutes) greater than 4 hrs. So, it is not certified.

@sailesh-ramayanam  Please see the changes now for the problem description.

**Review by sailesh-ramayanam — APPROVED** [review]

**Pull request merged**